test/test_nn.py

```diff
@@ -6519,18 +6519,32 @@ def test_noncontig_conv_grad_cuda(self, dtype=torch.float):
         self.assertEqual(result, input.grad.data, atol=dtype2prec_DONTUSE[dtype], rtol=0)
 
     def test_pixel_shuffle(self):
-        batch_size = random.randint(1, 3)
-        upscale_factor = random.randint(2, 5)
-        channels = random.randint(1, 4) * upscale_factor ** 2
-        height = random.randint(5, 10)
-        width = random.randint(5, 10)
-
-        input = torch.rand(batch_size, channels, height, width, requires_grad=True)
-        ps = nn.PixelShuffle(upscale_factor)
-        output = ps(input)
-        self._verify_pixel_shuffle(input.data, output.data, upscale_factor)
-        output.backward(output.data)
-        self.assertEqual(input.data, input.grad.data)
+        def _test_pixel_shuffle_helper(num_batch_dims=1):
+            batch_sizes = [random.randint(1, 3) for _ in range(num_batch_dims)]
+            upscale_factor = random.randint(2, 5)
+            channels = random.randint(1, 4) * upscale_factor ** 2
+            height = random.randint(5, 10)
+            width = random.randint(5, 10)
+
+            input = torch.rand(*batch_sizes, channels, height, width, requires_grad=True)
+            ps = nn.PixelShuffle(upscale_factor)
+            output = ps(input)
+            self._verify_pixel_shuffle(input.data, output.data, upscale_factor)
+            output.backward(output.data)
+            self.assertEqual(input.data, input.grad.data)
```

Comment:
This is an interesting autograd test. I don't think this PR needs to change it, though.